scalene/scalene_profiler.py

```diff
@@ -937,12 +937,9 @@ def enter_function_meta(
         f = frame
         try:
             while "<" in Filename(f.f_code.co_name):
-                f = cast(FrameType, frame.f_back)
-                if (
-                    "<genexpr>" in f.f_code.co_name
-                    or "<module>" in f.f_code.co_name
-                    or "<listcomp>" in f.f_code.co_name
-                ):
+                f = cast(FrameType, f.f_back)
+                # Handle case where the <\w+> is at the bottom of the stack
```

Comment:
This comment + regexp is pretty opaque. How about "where the function named <whatever> ...".